src/style/themeManager.ts

```diff
@@ -1,9 +1,12 @@
 import _ from 'lodash';
+import type {Context} from 'react';
 import Colors from './colors';
 
 
 export class ThemeManager {
```

Comment:
Not sure if you've tested this in private's node_modules, but it might be possible to do:
```
ThemeManager<ContextType = any>
```
(and change the `any`s to `ContextType`)
Which might make usage prettier.

Reply:
But i'm not instantiating ThemeManager in private, but in public. 
We kinda export it as a singleton instance, so I can't really pass a generics

Reply:
I did not test it, but I think it might affect the usage of `ThemeManager.getThemeContext()`, it might give us autocomplete with the generics added (used as `ThemeManager<ThemeContextProps>.getThemeContext()`, I think :S ).

Reply:
Not sure I understand, how would you use that exactly? In private for example?

Reply:
I assume that the getter needs to be used in private.
We can postpone this so it's easier to test.